.github/workflows/build.yml

```diff
@@ -15,6 +15,8 @@ jobs:
 
     steps:
     - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0 # avoid shallow clone so nbgv can do its work.
```

Comment:
Does it need to clone the entire history to work? Can it be truncated to a reasonable upper bound perhaps?

Reply:
It only needs to be high enough to reach the last change to version.json. But in practice, that's a hard number to estimate, and unless you're in a repo like VS, shallow clones don't really save significant time on clone in my experience.